### PR TITLE
CleanUp createUfsBlockReader

### PR DESCRIPTION
**Pull request by jja725**
### What changes are proposed in this pull request?

Remove mBlockIdToSessionIds since USER_UFS_BLOCK_READ_CONCURRENCY_MAX is broken. 
Merge openUFSBlock into createUfsBlockReader.
Remove redundant test cases since they are basically duplicates of UnderFileSystemBlockStoreTest
### Why are the changes needed?

Previously, mBlockIdToSessionIds is introduced to control UfsReadConcurrency. But basically, this functionality is broken because when we `createUfsBlockReader` we just drop the value returned from `openUfsBlock`(line 511 in `DefaultBlockWorker`) and create BlockReader anyway. 
And IMO USER_UFS_BLOCK_READ_CONCURRENCY_MAX is not a good way to control the concurrency to UFS(the default value is Integer.MAX_VALUE which is almost restricting nothing). 
So we propose to delete this broken functionality instead of picking up the dropped value. later on, we need to reimplement this throttling to UFS in a proper way.

Merge openUFSBlock into createUfsBlockReader since openUFSBlock is always used within createUfsBlockReader and openUFSBlock is just a combination of UFS fallback logic and acquire access(the name is confusing as well since it does not actually open block).

### Does this PR introduce any user-facing changes?
na

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java [hunk 1]
```diff
@@ -508,7 +508,14 @@ public BlockReader createUfsBlockReader(long sessionId, long blockId, long offse
       boolean positionShort, Protocol.OpenUfsBlockOptions options)
       throws BlockDoesNotExistException, IOException {
     try {
-      openUfsBlock(sessionId, blockId, options);
+      if (!options.hasUfsPath() && options.hasBlockInUfsTier() && options.getBlockInUfsTier()) {
```
Comment (rongrong) [inline core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java:511]: I was just looking at this today. The `options.hasBlockInUfsTier()` should not be necessary because `block_in_ufs_tier` is a boolean and the default value is `false`. Actually this logic of changing options should be moved to `createBlockReader`. Please take a look at https://github.com/Alluxio/alluxio/pull/15382 for related changes.

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java [hunk 2]
```diff
@@ -508,7 +508,14 @@ public BlockReader createUfsBlockReader(long sessionId, long blockId, long offse
...
+        // This is a fallback UFS block read. Reset the UFS block path according to the UfsBlock
+        // flag.
+        UfsManager.UfsClient ufsClient = mUfsManager.get(options.getMountId());
+        options = options.toBuilder()
+            .setUfsPath(alluxio.worker.BlockUtils.getUfsBlockPath(ufsClient, blockId)).build();
+      }
+      mUnderFileSystemBlockStore.acquireAccess(sessionId, blockId, options);
```
Comment (rongrong) [inline core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java:518]: We can move this to `UnderFileSystemBlockStore.createBlockReader` and make `acquireAccess` private.

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 3]
```diff
@@ -108,33 +108,6 @@ public void before() throws IOException {
         mSessions, mBlockStore, mUfsManager);
   }
 
-  @Test
-  public void openUnderFileSystemBlock() throws Exception {
```
Comment (rongrong) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:112]: Since the logic is moved to `createUfsBlockReader`, make sure the logic is still tested.

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java (hunk 1 above)
Comment (jja725) [reply core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java:511]: We can merge your PR first so we can adopt the change.

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java (hunk 3 above)
Comment (jja725) [reply core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:112]: the test here is covered by UnderFileSystemBlockStoreTest

File: core/server/worker/src/test/java/alluxio/worker/block/UnderFileSystemBlockStoreTest.java [hunk 4]
```diff
@@ -50,8 +49,6 @@ public void acquireAccess() throws Exception {
     for (int i = 0; i < 5; i++) {
       assertTrue(blockStore.acquireAccess(i + 1, BLOCK_ID, mOpenUfsBlockOptions));
     }
-
-    assertFalse(blockStore.acquireAccess(6, BLOCK_ID, mOpenUfsBlockOptions));
```
Comment (rongrong) [inline core/server/worker/src/test/java/alluxio/worker/block/UnderFileSystemBlockStoreTest.java:54]: Why is this removed?

**Review by rongrong — APPROVED** [review]

File: core/server/worker/src/test/java/alluxio/worker/block/UnderFileSystemBlockStoreTest.java (hunk 4 above)
Comment (jja725) [reply core/server/worker/src/test/java/alluxio/worker/block/UnderFileSystemBlockStoreTest.java:54]: This assertion is broken because we remove the UFS concurrency limit

**Comment by rongrong** [comment]
alluxio-bot, merge this please
